src/Pact/Types/Names.hs

```diff
@@ -230,17 +254,34 @@ nameParser i = (QName <$> qualifiedNameParser i <?> "qualifiedName") <|>
 
 
 instance Hashable Name where
-  hashWithSalt s (Name (BareName t _)) = s `hashWithSalt` (0::Int) `hashWithSalt` t
-  hashWithSalt s (QName (QualifiedName q n _)) = s `hashWithSalt` (1::Int) `hashWithSalt` q `hashWithSalt` n
+  hashWithSalt s (Name (BareName t _)) =
+    s `hashWithSalt` (0::Int) `hashWithSalt` t
+  hashWithSalt s (QName (QualifiedName q n _)) =
+    s `hashWithSalt` (1::Int) `hashWithSalt` q `hashWithSalt` n
+  hashWithSalt s (DName DynamicName{..}) =
+    s `hashWithSalt` (2::Int) `hashWithSalt` _dynMember
+    `hashWithSalt` _dynRefArg `hashWithSalt` _dynInterfaces
 instance Eq Name where
-  (QName (QualifiedName a b _)) == (QName (QualifiedName c d _)) = (a,b) == (c,d)
-  (Name (BareName a _)) == (Name (BareName b _)) = a == b
+  (QName (QualifiedName a b _)) == (QName (QualifiedName c d _)) =
```

Comment:
Sigh ... we need quickchecking of `Eq` and `Ord` guarantees or something, it's worrisome that this blob of code is untested.